checker/src/tag_domain.rs

```diff
@@ -17,12 +18,80 @@ macro_rules! does_propagate_tag {
     };
 }
 
+/// A replication of the `DefId` type from rustc. The type is used to implement serialization.
```

Comment:
I have avoided serializing DefId structures because they are compilation specific and the intent of the serializer is to produce summaries that can be used across compiles. For example the summaries produced by running the compiler over the standard library are used by subsequent compilations on completely different environments.

However, performance considerations and limitations on concurrent access to the persistent summary store has led to a situation where the persistent summary store is only used for standard library summaries, which may be complete overkill.

I expect to remove the persistent summary store at some point. Meanwhile, the erroneous serialization of the def ids you introduce here will be OK as long as no summaries in the persistent store contain them.

Reply:
I see. Do you have an idea what could be a good representation of tag types in the analyzer?

Reply:
You can stick with the DefId. Elsewhere ExpressionType is used, which is not ideal and certainly not appropriate for tags.